Mathlib/Geometry/RingedSpace/LocallyRingedSpace/ResidueField.lean

```diff
@@ -0,0 +1,113 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.Geometry.RingedSpace.LocallyRingedSpace
+
+/-!
+
+# Residue fields of points
+
+Any point `x` of a locally ringed space `X` comes with a natural residue field, namely the residue
+field of the stalk at `x`. Moreover, for every open subset of `X` containing `x`, we have a
+canonical evaluation map from `Γ(X, U)` to the residue field of `X` at `x`.
+
+## Main definitions
+
+The following are in the `AlgebraicGeometry.LocallyRingedSpace` namespace:
+
+- `residueField`: the residue field of the stalk at `x`.
+- `evaluation`: for open subsets `U` of `X` containing `x`, the evaluation map from sections over
+  `U` to the residue field at `x`.
+- `evaluationMap`: a morphism of locally ringed spaces induces a morphism, i.e. extension, of
+  residue fields.
+
+-/
+
+universe u
+
+open CategoryTheory TopologicalSpace Opposite
+
+noncomputable section
+
+namespace AlgebraicGeometry.LocallyRingedSpace
+
+variable (X : LocallyRingedSpace.{u}) {U : Opens X}
+
+/-- The residue field of `X` at a point `x` is the residue field of the stalk of `X`
+at `x`. -/
+def residueField (x : X) : CommRingCat :=
+  CommRingCat.of <| LocalRing.ResidueField (X.stalk x)
+
+instance (x : X) : Field (X.residueField x) :=
+  inferInstanceAs <| Field (LocalRing.ResidueField (X.stalk x))
+
+/--
+If `U` is an open of `X` containing `x`, we have a canonical ring map from the sections
+over `U` to the residue field of `x`.
+
+If we interpret sections over `U` as functions of `X` defined on `U`, then this ring map
+corresponds to evaluation at `x`.
+-/
+def evaluation (x : U) : X.presheaf.obj (op U) ⟶ X.residueField x :=
+  X.presheaf.germ x ≫ LocalRing.residue _
+
+/-- The global evaluation map from `Γ(X, ⊤)` to the residue field at `x`. -/
+def Γevaluation (x : X) : X.presheaf.obj (op ⊤) ⟶ X.residueField x :=
+  X.evaluation ⟨x, show x ∈ ⊤ from trivial⟩
+
+@[simp]
```

Comment:
Do we really want this to be a simp lemma? (same for `Γevaluation_eq_zero_iff_not_mem_basicOpen`.)

Reply:
I think we should choose a simpNF. Do you think `X.evaluation x f = 0` would be better?

Reply:
Ok, let us keep these simp lemmas.